logbook/oplog/log.go

```diff
@@ -92,6 +93,68 @@ type Logstore interface {
 	Descendants(ctx context.Context, l *Log) error
 }
 
+// SparseAncestorsAllDescendantsLogstore is a an extension interface to
+// Logstore with an optimized method for getting a log with sparse parents and
+// all descendants
+type SparseAncestorsAllDescendantsLogstore interface {
```

Comment:
Not a huge fan of these names, because they're talking about implementation details of what is being returned, rather than the intention of why a caller would want to use them. That said, I think the reason they should be used is that we don't really have the vocabulary for what they should be called instead, and hopefully that sort of question will be answered in future design work. This is fine for now, but we should keep in mind that we need to come up with some name for what's really going on here when deciding on future logbook interfaces.

Reply:
totally agreed. I've intentionally made it something really idiosyncratic so it won't be a name we'll want to use. That said, It's also part of the oplog package, so I'm ok with obtuse names. The logbook interface is more important for qri, but this one definitely needs more thought